thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -743,13 +743,12 @@ class CIndexReadSlaveActivity : public CIndexReadSlaveBase
         if (!helper->transformMayFilter() && !helper->hasMatchFilter())
         {
             if (choosenLimit)
-            {
                 remoteLimit = choosenLimit;
-                if ((RCMAX != keyedLimit) && (keyedLimit+1 < remoteLimit))
-                    remoteLimit = keyedLimit+1; // 1 more to ensure triggered when received back.
-            }
         }
 
+        if ((RCMAX != keyedLimit) && (keyedLimit+1 < remoteLimit))
+            remoteLimit = keyedLimit+1; // 1 more to ensure triggered when received back.
```

Comment:
If choosenLimit is less than this, you may not want to increase it? Not sure what the semantics are in such cases.

Reply:
I'm not sure either, I thought even if choosen would prevent the output exceeding the keyedLimit,
the semantics of the keyed limit meant that it could be exceeded and cause the limit action (e.g. exception).

Reply:
@ghalliday - I think this is correct. keyedLimit can be higher and be hit, before choosenLimit is reached.
So I think it's correct that when there's a keyedLimit, remoteLimit is set to keyedLimit+1.

Reply:
Really you should be passing the keyed limit to the remote end separately from the choosen, and checking it there.

Reply:
I agree, I've opened a JIRA : https://track.hpccsystems.com/browse/HPCC-21546